pkg/backend/httpstate/state.go

```diff
@@ -180,6 +181,11 @@ func (b *cloudBackend) newUpdate(ctx context.Context, stackRef backend.StackRefe
 			tok, err := b.Client().RenewUpdateLease(
 				ctx, update, currentToken, duration)
 			if err != nil {
+				// Translate 403 status codes to expired token errors to stop the token refresh loop.
+				var apierr apitype.ErrorResponse
+				if errors.As(err, &apierr) && apierr.Code == 403 {
+					return "", time.Time{}, expiredTokenError{err}
+				}
```

Comment:
I was wondering if a 403 is necessarily an expired token always, e.g. it could happen if the user sets a wrong token, but it looks like this loop is only used to renew the lease, so maybe it's fine to just always consider it expired?

Reply:
Yeh if it's a bad token to start with we'll have never started the loop